src/main/java/org/matsim/pt2matsim/mapping/networkRouter/ScheduleRoutersStandard.java

```diff
@@ -79,26 +85,56 @@ private void load() {
 		log.info("==============================================");
 		log.info("Creating network routers for transit routes...");
 		log.info("Initiating network and router for transit routes...");
-		LeastCostPathCalculatorFactory factory = new SpeedyALTFactory();
-		for(TransitLine transitLine : schedule.getTransitLines().values()) {
-			for(TransitRoute transitRoute : transitLine.getRoutes().values()) {
-				String scheduleMode = transitRoute.getTransportMode();
-				PathCalculator tmpRouter = pathCalculatorsByMode.get(scheduleMode);
-				if(tmpRouter == null) {
-					log.info("New router for schedule mode " + scheduleMode);
-					Set<String> networkTransportModes = transportModeAssignment.get(scheduleMode);
+		
+		
+		if (this.networkRouter.equals(RoutingAlgorithmType.SpeedyALT)) {
+			LeastCostPathCalculatorFactory factory = new SpeedyALTFactory();
+			for(TransitLine transitLine : schedule.getTransitLines().values()) {
+				for(TransitRoute transitRoute : transitLine.getRoutes().values()) {
+					String scheduleMode = transitRoute.getTransportMode();
+					PathCalculator tmpRouter = pathCalculatorsByMode.get(scheduleMode);
+					if(tmpRouter == null) {
+						log.info("New router for schedule mode " + scheduleMode);
+						Set<String> networkTransportModes = transportModeAssignment.get(scheduleMode);
+
+						Network filteredNetwork = NetworkTools.createFilteredNetworkByLinkMode(this.network, networkTransportModes);
+
+						LocalRouter r = new LocalRouter();
+
+						tmpRouter = new PathCalculator(factory.createPathCalculator(filteredNetwork, r, r));
 
-					Network filteredNetwork = NetworkTools.createFilteredNetworkByLinkMode(this.network, networkTransportModes);
+						pathCalculatorsByMode.put(scheduleMode, tmpRouter);
+						networksByMode.put(scheduleMode, filteredNetwork);
+					}
+				}
+			}
+		}
+		
+		else if (this.networkRouter.equals(RoutingAlgorithmType.AStarLandmarks)) {
+			LeastCostPathCalculatorFactory factory = new AStarLandmarksFactory(nThreads);
+			for(TransitLine transitLine : schedule.getTransitLines().values()) {
```

Comment:
This code inside the if statement looks duplicated to me, is there a reason to it?

Reply:
Thank you for noticing. No good reason, it is now removed in [bcaa82c](https://github.com/matsim-org/pt2matsim/pull/284/commits/bcaa82ce2a899220ce2a17344d0df258babd9db7)